### PR TITLE
add recipes for Broadway

### PR DESCRIPTION
**Pull request by othillo**
| Q             | A
| ------------- | ---
| License       | MIT

### CODE REVIEW

File: broadway/event-store-dbal/0.1/config/packages/broadway_event_store_dbal.yaml [hunk 1]
```diff
@@ -1,7 +0,0 @@
-broadway:
```
Comment (Nyholm) [inline broadway/event-store-dbal/0.1/config/packages/broadway_event_store_dbal.yaml]: There is no need to remove old recipes.

File: broadway/event-store-dbal/0.2/config/packages/broadway_event_store_dbal.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+broadway:
+    event_store: broadway.event_store.dbal
+
+services:
+    broadway.event_store.dbal:
+        class: Broadway\EventStore\Dbal\DBALEventStore
+        public: true
```
Comment (Nyholm) [inline broadway/event-store-dbal/0.2/config/packages/broadway_event_store_dbal.yaml]: Do you really want this to be public by default?

File: broadway/event-store-dbal/0.2/config/packages/broadway_event_store_dbal.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+broadway:
+    event_store: broadway.event_store.dbal
+
+services:
+    broadway.event_store.dbal:
```
Comment (Nyholm) [inline broadway/event-store-dbal/0.2/config/packages/broadway_event_store_dbal.yaml]: Let's use `Broadway\EventStore\Dbal\DBALEventStore` as service id.

File: broadway/read-model-mongodb/0.2/manifest.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+    "copy-from-recipe": {
+        "config/": "%CONFIG_DIR%/"
+    },
+    "container": {
+        "mongodb_host": "localhost",
+        "mongodb_port": "27107",
```
Comment (Nyholm) [inline broadway/read-model-mongodb/0.2/manifest.json]: These parameters build up a DSN. We should consider using the DSN here directly. That would reduce the number of parameters.

**Comment by othillo** [comment]
I'm a little confused about the FQCN usage. I understand it helps auto-wiring but then again Symfony suggests public bundles not to rely on auto-wiring (https://symfony.com/doc/current/service_container/autowiring.html#public-and-reusable-bundles). So for now I don't see the benefit, but, please, correct me if I'm wrong.

**Comment by nicolas-grekas** [comment]
A service should get FQCN id only if it should be autowireable. That's the rule.

**Comment by Nyholm** [comment]
Thank you. You are both correct. I thought one should always use FQCN no matter auto-wiring.

**Comment by othillo** [comment]
@Nyholm I've done the changes you requested. Can you review it again?

File: broadway/event-store-mongodb/0.2/manifest.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{
+    "copy-from-recipe": {
+        "config/": "%CONFIG_DIR%/"
+    },
+    "container": {
+        "mongodb_uri": "mongodb://127.0.0.1/",
+        "mongodb_database": "default"
```
Comment (Nyholm) [inline broadway/event-store-mongodb/0.2/manifest.json]: I assume these will be different on different environments. Lets put them as environment variables: https://github.com/symfony/recipes#env-configurator

File: broadway/read-model-mongodb/0.2/manifest.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{
+    "copy-from-recipe": {
+        "config/": "%CONFIG_DIR%/"
+    },
+    "container": {
+        "mongodb_uri": "mongodb://127.0.0.1/",
+        "mongodb_database": "default"
```
Comment (Nyholm) [inline broadway/read-model-mongodb/0.2/manifest.json]: Same here. Consider using environment variables. 

Im not sure what happends when you reuse environment variables. Lets try =)

**Comment by othillo** [comment]
I introduced environment variables for MongoDB. The Travis builds are failing but I don't understand why.

File: broadway/read-model-elasticsearch/0.3/config/packages/broadway_read_model_elasticsearch.yaml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+        class: Broadway\ReadModel\ElasticSearch\ElasticSearchRepositoryFactory
+        arguments:
+            - "@broadway.elasticsearch.client"
+            - "@broadway.serializer.readmodel"
+
+    broadway.elasticsearch.client:
+        class: Elasticsearch\Client
+        factory: ['@broadway.elasticsearch.client_factory', create]
+        arguments:
+            - "%elasticsearch%"
```
Comment (Nyholm) [inline broadway/read-model-elasticsearch/0.3/config/packages/broadway_read_model_elasticsearch.yaml]: This parameter is undefined. You should consider put it as an environment variable.

Comment (Nyholm) [reply broadway/read-model-elasticsearch/0.3/config/packages/broadway_read_model_elasticsearch.yaml]: Ping. Could you fix this and the PR is all good.

Comment (othillo) [reply broadway/read-model-elasticsearch/0.3/config/packages/broadway_read_model_elasticsearch.yaml]: Unfortunately this parameter is a nullable array so we cannot use environment variables here. I declared it and set it to null by default.

Comment (othillo) [reply broadway/read-model-elasticsearch/0.3/config/packages/broadway_read_model_elasticsearch.yaml]: @Nyholm are you ok with this?

**Comment by Nyholm** [comment]
Excellent. Thank you

**Pull request merged**